pkg/syslog/client.go

```diff
@@ -0,0 +1,44 @@
+package syslog
+
+import (
+	"net"
+
+	"github.com/flynn/flynn/pkg/syslog/rfc5424"
+	"github.com/flynn/flynn/pkg/syslog/rfc6587"
+)
+
+type Client struct {
+	c net.Conn
+
+	h *rfc5424.Header
+	w *rfc5424.Writer
+}
+
+func Dial(address string, hdr *rfc5424.Header) (*Client, error) {
+	conn, err := net.Dial("tcp", address)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		c: conn,
+		h: hdr,
+		w: rfc5424.NewWriter(rfc6587.NewWriter(conn)),
+	}, nil
+}
+
+func (c *Client) Close() error {
+	return c.c.Close()
+}
+
+func (c *Client) Write(b []byte) (int, error) {
+	return c.WriteFormat(c.h, nil, b)
+}
+
+func (c *Client) WriteFormat(hdr *rfc5424.Header, sd rfc5424.StructuredData, msg []byte) (int, error) {
```

Comment:
Doesn't look like this is used anywhere other `Write()` above. Is there a reason it needs to be a separate function, especially exported?